configBackup.go

```diff
@@ -278,3 +207,49 @@ func (config *configurationFile) loadConfig(verboseFlag bool, debugFlag bool) er
 
 	return err
 }
+
+func readSection(viper *viper.Viper, sectionKey string) []map[string]string {
+	if viper.IsSet(sectionKey) {
+		section := make([]interface{}, 0)
+		if viper.IsSet(sectionKey + ".1") {
+			// Keyed by number
+			for i := 1; ; i++ {
+				key := sectionKey + "." + strconv.Itoa(i)
+				if viper.IsSet(key) {
+					section = append(section, viper.Get(key))
+				} else {
+					break
+				}
+			}
+		} else {
+			// Array
+			section = viper.Get(sectionKey).([]interface{})
+		}
+
+		return coerceToArrayOfMapStringString(section[0:])
+	}
+	return nil
+}
+
+func coerceToArrayOfMapStringString(slice []interface{}) []map[string]string {
+	section := make([]map[string]string, len(slice))
+	for i, item := range slice {
+		itemMap := make(map[string]string)
+		switch item.(type) {
+		case map[string]interface{}:
+			for key, value := range item.(map[string]interface{}) {
+				strKey := key
+				strValue := fmt.Sprintf("%v", value)
+				itemMap[strKey] = strValue
+			}
+		case map[interface{}]interface{}:
```

Comment:
This is the basis of where I got into trouble with Viper, I think. In the case of arrays, you don't NEED to coerce to an array of `map[string]string` (you get it that way), but you do when we read the section this way.

So coercing is one way? The other way, I presume, would be to read in a config format (directly into a structure)? Or would you still need this then? Or are you not sure because you didn't look at that?

What, exactly, are you testing for here? I can see what the code is doing, but I don't fully understand what is being tested here. I approve the changes (they work fine under test), I'd just like to fully understand them. Thanks!

Reply:
If you're asking about the case statement, when reading via the keys (1:, 2:, etc) it is a type of `map[string]interface{}`, but when reading via the new method of a list it is a type of `map[interface{}]interface{}`. 

If you remove the older key based format that part of the `coerceToArrayOfMapStringString` can be removed as it won't be needed anymore.


I'm new to golang, and intuitively it would seem you could use the more generic `map[interface{}]interface{}` for both, but in my testing that was not the case and using a case method like this is how I had seen it done in snippets elsewhere.

Reply:
Got it. So after we drop support for the old version, you only need the cast at line 245, and the case at line 239 is no longer necessary. Right?

Reply:
BTW, FWIW: Right now, you can specify invalid fields, and as long as valid fields are specified, this will pass as a valid file. So you can say something like: `foo: bar`, and I think the existing code would take that. Not only that but, as I recall, the old code would not take that since we tended to do our checking on a field by field basis. So, technically, this is a regression.

I think it makes sense to have a list of mandatory parameters and optional parameters for each of backup, copy, prune, and check operations. If you specify a parameter that isn't in the mandatory or optional list, then it gets rejected. This keeps you from specifying garbage in your configuration file. And if the entire mandatory list isn't in your result, then it gets rejected as well.

I'll put this on the "to-do" list. Not hard to do, but something we (either you or me) should do.

Reply:
The previous design also would allow invalid keys, it only checked for the expected keys and ignored any invalid ones, same as this design.

Reply:
I just checked. Oh good, you're right! 💯  It's actually easier now to check for invalid values; before, it would just be any value in the viper file. Now, it ends up in a map, and that's easier to see what is defined.

Any thought on the prior comment? To drop support for the old (numbered) sections, we just get rid of the code that reads it in and then get rid of the case at line 239, right?

Reply:
Thanks! I already added a deprecation notice and merged to master. I suggest you update to the master branch, look over what I added, and make sure you're satisfied with it. I put explanations in my commit messages, where appropriate, to try and explain why I made the changes. If anything isn't clear, feel free to ask. Thanks again for your contribution to the project!